code/game/machinery/deployable.dm

```diff
@@ -158,12 +163,53 @@
 	addtimer(CALLBACK(src, PROC_REF(deploy)), deploy_time)
 
 /obj/structure/barricade/security/proc/deploy()
-	icon_state = "barrier1"
-	set_density(TRUE)
+	toggle_lock() //monkestation edit
 	set_anchored(TRUE)
 	if(deploy_message)
 		visible_message(span_warning("[src] deploys!"))
 
+//MONKESTATION EDIT START
+/obj/structure/barricade/security/proc/toggle_lock()
+	if(!locked)
+		set_density(TRUE)
+		icon_state = "barrier1"
+		locked = TRUE
+		playsound(src, 'sound/machines/boltsup.ogg', 30)
+	else
+		set_density(FALSE)
+		icon_state = "barrier0"
+		locked = FALSE
+		playsound(src, 'sound/machines/boltsdown.ogg', 30)
+	update_appearance()
+
+/obj/structure/barricade/security/attackby(obj/item/tool, mob/living/user, params)
+	if(isidcard(tool))
+		var/obj/item/card/id/id_card = tool
+		if((ACCESS_SECURITY in id_card.GetAccess()))
+			toggle_lock()
+			balloon_alert(user, "barrier [locked ? "locked" : "unlocked"]")
+		else
+			balloon_alert(user, "no access!")
+	else
+		return ..()
```

Comment:
This could be better organized
```suggestion
	if(!isidcard(tool))
		return ..()
		
	var/obj/item/card/id/id_card = tool
	if(!(ACCESS_SECURITY in id_card.GetAccess()))
		balloon_alert(user, "no access!")
	
	toggle_lock()
	balloon_alert(user, "barrier [locked ? "locked" : "unlocked"]")
```